test/e2e/instrumentation/monitoring/custom_metrics_deployments.go

```diff
@@ -252,7 +288,19 @@ func CleanupDescriptors(service *gcm.Service, projectId string) {
 
 // CleanupAdapter deletes Custom Metrics - Stackdriver adapter deployments.
 func CleanupAdapter() error {
-	stat, err := framework.RunKubectl("delete", "-f", "https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/adapter-beta.yaml")
+	return CleanupAdapterForOldResourceModel()
+}
+
+// CleanupAdapterForOldResourceModel deletes Custom Metrics - Stackdriver adapter deployments.
+func CleanupAdapterForOldResourceModel() error {
+	stat, err := framework.RunKubectl("delete", "-f", "https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/deploy/staging/adapter_old_resource_model.yaml")
+	framework.Logf(stat)
+	return err
+}
+
+// CleanupAdapterForNewResourceModel deletes Custom Metrics - Stackdriver adapter deployments.
+func CleanupAdapterForNewResourceModel() error {
+	stat, err := framework.RunKubectl("delete", "-f", "https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/deploy/staging/adapter_new_resource_model.yaml")
```

Comment:
Can you put this yaml in a variable somewhere during creation? This can fail to delete the object e.g. when it gets updated on github. Same above.

Reply:
I prefer not to do this, it would introduce more complication to this code - right now it uses framework function 'runKubectl', and it would have to download the deployment, store it in some variable in the test code, and use "echo %q | kubectl create -f -" to create it. The impact is close to none - a deployment update potentially affects only a single run of e2e tests and only for Custom Metrics - Stackdriver Adapter. We don't look at individual failures, since this happens occasionally anyway. WDYT?

If you are not convinced, what about adding a TODO for now, since it doesn't introduce a regression? I see some benefits, but I'm not yet sure how exactly to do this, so I prefer to fix this in a follow up PR.

Reply:
Maybe it could be stored in a file locally, but if you prefer authoring a TODO instead then I'm not going to block this.

Reply:
Done (actually implemented it storing file locally).